MonoGame.Framework/GraphicsDeviceManager.cs

```diff
@@ -280,20 +280,193 @@ private void PreparePresentationParameters(PresentationParameters presentationPa
             PlatformPreparePresentationParameters(presentationParameters);
         }
 
-        private void PrepareGraphicsDeviceInformation(GraphicsDeviceInformation gdi)
+        /// <summary>
+        /// Finds the best device configuration that is compatible with the current device preferences.
+        /// </summary>
+        /// <param name="anySuitableDevice">If true, the best device configuration can be selected from any
+        /// available adaptor. If false, only the current adaptor is used.</param>
+        /// <returns>The best device configuration that could be found.</returns>
+        protected virtual GraphicsDeviceInformation FindBestDevice(bool anySuitableDevice)
+        {
+            // Create a list of available devices
+            var devices = new List<GraphicsDeviceInformation>();
+            if (anySuitableDevice)
+            {
+                foreach (var adapter in GraphicsAdapter.Adapters)
+                {
+                    if (adapter.IsProfileSupported(GraphicsProfile))
+                        AddModes(adapter, devices);
+                }
+            }
+            else
+            {
+                var adapter = GraphicsAdapter.DefaultAdapter;
+                if (adapter.IsProfileSupported(GraphicsProfile))
+                    AddModes(adapter, devices);
+            }
+
+            // Rank them to get the most preferred device first
+            RankDevices(devices);
+
+            // No devices left in the list?
+            if (devices.Count == 0)
+            {
+                throw new NoSuitableGraphicsDeviceException(FrameworkResources.CouldNotFindCompatibleGraphicsDevice);
+            }
+
+            // The first device in the list is the most suitable
+            return devices[0];
+        }
+
+        // Add all modes supported by an adapter to the list
+        void AddModes(GraphicsAdapter adapter, List<GraphicsDeviceInformation> devices)
         {
-            gdi.Adapter = GraphicsAdapter.DefaultAdapter;
-            gdi.GraphicsProfile = GraphicsProfile;
-            var pp = new PresentationParameters();
-            PreparePresentationParameters(pp);
-            gdi.PresentationParameters = pp;
+            int multiSampleCount = 0;
+            if (_preferMultiSampling)
+            {
+                // Always initialize MultiSampleCount to the maximum. If users want to overwrite
+                // this they have to respond to the PreparingDeviceSettingsEvent and modify
+                // args.GraphicsDeviceInformation.PresentationParameters.MultiSampleCount
+                multiSampleCount = GraphicsDevice != null ? GraphicsDevice.GraphicsCapabilities.MaxMultiSampleCount : 32;
```

Comment:
You could use GraphicsCapabilities.MultiSampleCountLimit instead of hardcoding 32. 
https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/GraphicsCapabilities.cs#L87
It's currently private.